### PR TITLE
cinny-desktop: fix build failure

### LINKED ISSUE #357406: Build failure: cinny-desktop
## Steps To Reproduce

1. build cinny-desktop

## Build log

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/b8fly5g7gba7fnw1s83am9v1c5rgp3wd-source
source root is source/src-tauri
Executing cargoSetupPostUnpackHook
unpacking source archive /nix/store/hrca7s044i3v9apn5w33mgxl36p7xird-cinny-desktop-4.2.3-vendor.tar.gz
Finished cargoSetupPostUnpackHook
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
substituteStream() in derivation cinny-desktop-4.2.3: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'tauri.conf.json')
substituteInPlace called without any files to operate on (files must come before options!)
```

</details>

## Additional context
This package doesn't seem to be checked by CI because of its security vuln in libolm. It seems they may be working towards fixing that, but for now, it is what it is I feel.

~~Also there seems to be an issue open for already existing problems with this package:~~ irrelevant issue

## Notify maintainers

@ryand56 @Qyriad 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Noodlez1232**
Unfortunately this leads to the loss of the system tray icon and the functionality that comes with that. I was unfortunately unable to figure out why this was the case, but in response, I have removed the dependencies that were included for it.

Hopefully somebody more experienced than me can fix it.

Fixes #357406

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ci/cinny-desktop/package.nix
Comment (getchoo) [inline pkgs/by-name/ci/cinny-desktop/package.nix:58]:
```suggestion
```
It'd be better to remove this

File: pkgs/by-name/ci/cinny-desktop/package.nix
Comment (getchoo) [inline pkgs/by-name/ci/cinny-desktop/package.nix:76]:
```suggestion
      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ libayatana-appindicator ]}" \
```
Uncommenting this *should* make appindicator work without the above patch

File: pkgs/by-name/ci/cinny-desktop/package.nix
Comment (getchoo) [inline pkgs/by-name/ci/cinny-desktop/package.nix:84]:
```suggestion
    makeBinaryWrapper
```
This can also remove a bit of overhead from the script-based wrapper

File: pkgs/by-name/ci/cinny-desktop/package.nix
Comment (getchoo) [inline pkgs/by-name/ci/cinny-desktop/package.nix:95]:
```suggestion
```

**Review by getchoo — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ci/cinny-desktop/package.nix
Comment (Noodlez1232) [reply pkgs/by-name/ci/cinny-desktop/package.nix:76]: I tested this already, which is why it's commented out. I'll just remove the commented parts.

**Comment by Noodlez1232** [comment]
I think it'd be a good idea to get approval from the maintainer @ryand56

**Review by ryand56 — APPROVED** [review]
I would prefer to have the tray icon still, but this will have to do for now.

**Pull request merged**
